### ISSUE TITLE
Replace custom time handling with datetime/timedelta

### ISSUE THREAD

**Issue by jacalata** [tableau/server-client-python#1336]
From #1299: 

The use of a custom datetime module caught me off guard in the project.
I'd suggest using datetime and timedelta as the source for seconds and minutes instead of manually keeping track of what the seconds and minutes ought to be, but didn't want to make any more changes than necessary to fix the problem.

**Comment by jacalata** [comment]
Would be nice to have — PRs welcome to swap the raw `time.time()` / `time.sleep()` in `exponential_backoff.py` for `datetime.timedelta`.

**Comment by jacalata** [comment]
Adding a datapoint from the recent cloud-schedule parsing work in #1875: `datetime_helpers.parse_datetime` had to grow a second `strptime` branch to accept the Cloud form `2026-08-29T16:55:00-0700` alongside Server's `...Z`.

`datetime.fromisoformat` would collapse both into a one-liner — but not until we drop 3.10:
- Python **3.11+** accepts both `...Z` and unpunctuated offsets like `-0700` natively.
- Python **3.10** rejects `Z` and requires punctuated offsets (`-07:00`), so the swap would still need a preprocessing shim — no cleaner than the current two-strptime pattern.

Once `requires-python` bumps past 3.10 (EOL 2026-10), `parse_datetime` becomes `datetime.fromisoformat(s)` and `TABLEAU_CLOUD_DATE_FORMAT` / the `try/except ValueError` fallback can be deleted. Flagging so it doesn't get lost.

**Comment by jacalata** [comment]
Closing as won't-fix rather than stale. On closer inspection (see #1876) the refactor turns out to be cosmetic — `(datetime.now() - self.start_time).total_seconds()` works the same way as `time.time() - self.start_time`, nothing downstream uses timedelta types, both APIs are equally subject to wall-clock jumps. Sorry for the earlier "PRs welcome" mixed signal.
